imod/typing/__init__.py

```diff
@@ -9,5 +9,6 @@
 
 GridDataArray: TypeAlias = Union[xr.DataArray, xu.UgridDataArray]
 GridDataset: TypeAlias = Union[xr.Dataset, xu.UgridDataset]
+ScalarDataArray: TypeAlias = Union[xr.DataArray, xu.UgridDataArray]
```

Comment:
This is the same type as GridDataArray.
I would prefer ScalarAsDataArray to avoid confusion with dataarrays that have scalars for array elements.
You could also create a derived type of xr.DataArray which has a single element, and use that derived type in the TypeAlias.   
Is the UgriDataArray  option really necessary? This type seems to be used here for options like "xt3d_off" which are not really structured or unstructured.

Reply:
I renamed it to ScalarAsDataArray. UgridDataArray option is necessary, to as object types are preserved, for example when UgridDataset[scalar_variable] -> UgridDataArray